lang_php/parsing/cst_php.ml

```diff
@@ -142,6 +147,7 @@ type name =
 type hint_type =
   | Hint of name (* only self/parent, no static *) *
             type_args option
+  | HintScopeResolution of class_scope_resolution_ident * type_args option
```

Comment:
Do we really need this? Can't we just use `Hint` directly? Anyways, that's what we do later in `Ast_php_build`.

Reply:
But don't take it as a "must-do" comment... I just think it's worth considering.

Reply:
@IagoAbal So, when `self` is used as a type parameter, are you suggesting that it gets represented as the following in the CST:

```
XName [QI (Name ("self", tok))]
```

Reply:
yes I don't think you need to modify the CST. The CST already has some special nodes for Self and Parent (in name). The main problem was that ast_php.ml didn't have those nodes anymore, and so the information was lost later when converting to the generic AST. Adding Self/Parent to ast_php.ml should be enough.